atom/common/native_mate_converters/blink_converter.cc

```diff
@@ -176,9 +176,10 @@ bool Converter<blink::WebKeyboardEvent>::FromV8(
   out->setKeyIdentifierFromWindowsKeyCode();
   if ((out->type == blink::WebInputEvent::Char ||
        out->type == blink::WebInputEvent::RawKeyDown) &&
-      str.size() == 1) {
-    out->text[0] = str[0];
-    out->unmodifiedText[0] = str[0];
+      str.size() <= 2) {
+    base::string16 code = base::UTF8ToUTF16(str);
+    out->text[0] = code[0];
```

Comment:
A character can take 4 bytes at maximum (for now), we should just ignore the size and copy all characters to `text`:

```c++
  // Make sure to not read beyond the buffer in case some bad code doesn't
  // NULL-terminate it (this is called from plugins).
  size_t text_length_cap = WebKeyboardEvent::textLengthCap;
  base::string16 text16 = base::UTF8ToUTF16(text);

  memset(out->text, 0, text_length_cap);
  memset(out->unmodifiedText, 0, text_length_cap);
  for (size_t i = 0; i < std::min(text_length_cap, text16.size()); ++i)
    out->text[i] = text16[i];
```